com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs

```diff
@@ -0,0 +1,525 @@
+                    set.name = EditorGUI.DelayedTextField(rect, set.name, EditorStyles.boldLabel);
+                    if (EditorGUI.EndChangeCheck())
+                    {
+                        m_RenameSelectedBakingSet = false;
+
+                        // Rename profile asset to match name:
+                        set.profile.name = set.name;
+                        AssetDatabase.RenameAsset(AssetDatabase.GetAssetPath(set.profile), set.name);
+                    }
+                }
+                else
+                    EditorGUI.LabelField(rect, set.name, EditorStyles.boldLabel);
+            };
+            m_BakingSets.elementHeightCallback = _ => EditorGUIUtility.singleLineHeight;
+            m_BakingSets.onSelectCallback = OnBakingSetSelected;
+
+            m_BakingSets.onAddCallback = (list) =>
+            {
+                Undo.RegisterCompleteObjectUndo(sceneData.parentAsset, "Added new baking set");
+                sceneData.CreateNewBakingSet("New Baking Set");
+                OnBakingSetSelected(list);
+            };
+
+            m_BakingSets.onRemoveCallback = (list) =>
+            {
+                if (m_BakingSets.count == 1)
+                {
+                    EditorUtility.DisplayDialog("Can't delete baking set", "You can't delete the last Baking set. You need to have at least one.", "Ok");
+                    return;
+                }
+                if (EditorUtility.DisplayDialog("Delete selected baking set?", $"Do you really want to delete the baking set '{sceneData.bakingSets[list.index].name}'?", "Yes", "Cancel"))
+                {
+                    Undo.RegisterCompleteObjectUndo(sceneData.parentAsset, "Deleted baking set");
+                    ReorderableList.defaultBehaviours.DoRemoveButton(list);
+                }
+            };
+
+            m_BakingSets.index = Mathf.Clamp(EditorPrefs.GetInt(k_SelectedBakingSetKey, 0), 0, m_BakingSets.count - 1);
+
+            OnBakingSetSelected(m_BakingSets);
+        }
+
+        void RefreshAfterUndo()
+        {
+            InitializeBakingSetList();
+
+            OnBakingSetSelected(m_BakingSets);
+
+            Repaint();
+        }
+
+        void RefreshSceneAssets()
+        {
+            var sceneAssets = AssetDatabase.FindAssets("t:Scene", new string[] { "Assets/" });
+
+            m_ScenesInProject = sceneAssets.Select(s =>
+            {
+                var path = AssetDatabase.GUIDToAssetPath(s);
+                var asset = AssetDatabase.LoadAssetAtPath<SceneAsset>(path);
+                return new SceneData
+                {
+                    asset = asset,
+                    path = path,
+                    guid = s
+                };
+            }).ToList();
+        }
+
+        SceneData FindSceneData(string guid)
+        {
+            // TODO: replace m_ScenesInProject list by a dictionary
+            var data = m_ScenesInProject.FirstOrDefault(s => s.guid == guid);
```

Comment:
Do we want to this TODO before getting this through?

Reply:
I guess it's fine